modules/coin/rest/handlers.go

```diff
@@ -63,17 +63,34 @@ func doQueryAccount(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if err := proofs.OutputProof(account, proof.BlockHeight()); err != nil {
+	if err := proofs.FoutputProof(w, account, proof.BlockHeight()); err != nil {
 		common.WriteError(w, err)
-		return
 	}
-	common.WriteSuccess(w, account)
+}
+
+func PrepareSendTx(si *SendInput) basecoin.Tx {
```

Comment:
Okay separate the validation/parse for the construction.  Makes sense, also easier to test.